core/backend/src/test/standalone/InlineGeometryPartReferences.test.ts

```diff
@@ -0,0 +1,527 @@
+    imodel = SnapshotDb.createEmpty(IModelTestUtils.prepareOutputFile("InlineGeomParts", `${Guid.createValue()}.bim`), {
+      rootSubject: { name: "InlineGeomParts", description: "InlineGeomParts" },
+    });
+
+    GenericSchema.registerSchema();
+    const partitionId = imodel.elements.insertElement({
+      classFullName: PhysicalPartition.classFullName,
+      model: IModel.repositoryModelId,
+      parent: new SubjectOwnsPartitionElements(IModel.rootSubjectId),
+      code: PhysicalPartition.createCode(imodel, IModel.rootSubjectId, `PhysicalPartition_${Guid.createValue()}`),
+    });
+
+    expect(Id64.isValidId64(partitionId)).to.be.true;
+    const model = imodel.models.createModel({
+      classFullName: PhysicalModel.classFullName,
+      modeledElement: { id: partitionId },
+    });
+
+    expect(model).instanceOf(PhysicalModel);
+
+    modelId = imodel.models.insertModel(model.toJSON());
+    expect(Id64.isValidId64(modelId)).to.be.true;
+
+    categoryId = SpatialCategory.insert(imodel, IModel.dictionaryId, "ctgry", { color: ColorDef.blue.toJSON() });
+    expect(Id64.isValidId64(categoryId)).to.be.true;
+    blueSubCategoryId = IModel.getDefaultSubCategoryId(categoryId);
+    redSubCategoryId = SubCategory.insert(imodel, categoryId, "red", { color: ColorDef.red.toJSON() });
+    expect(Id64.isValidId64(redSubCategoryId)).to.be.true;
+
+    materialId = RenderMaterialElement.insert(imodel, IModel.dictionaryId, "mat", { paletteName: "pal" });
+    expect(Id64.isValidId64(materialId)).to.be.true;
+  });
+
+  afterEach(() => {
+    imodel.close();
+  });
+
+  function insertGeometryPart(geom: GeomWriterEntry[]): string {
+    const writer = new GeomWriter();
+    for (const entry of geom)
+      writer.append(entry);
+
+    const props: GeometryPartProps = {
+      classFullName: GeometryPart.classFullName,
+      model: IModel.dictionaryId,
+      code: GeometryPart.createCode(imodel, IModel.dictionaryId, Guid.createValue()),
+      geom: writer.builder.geometryStream,
+    };
+
+    const partId = imodel.elements.insertElement(props);
+    expect(Id64.isValidId64(partId)).to.be.true;
+    return partId;
+  }
+
+  function insertElement(geom: GeomWriterEntry[], viewIndependent = false): string {
+    const writer = new GeomWriter({ categoryId });
+    if (viewIndependent)
+      writer.builder.isViewIndependent = true;
+
+    for (const entry of geom)
+      writer.append(entry);
+
+    const props: GeometricElement3dProps = {
+      classFullName: PhysicalObject.classFullName,
+      model: modelId,
+      code: Code.createEmpty(),
+      category: categoryId,
+      geom: writer.builder.geometryStream,
+      placement: {
+        origin: [0, 0, 0],
+        angles: { },
+      },
+    };
+
+    const elemId = imodel.elements.insertElement(props);
+    expect(Id64.isValidId64(elemId)).to.be.true;
+    return elemId;
+  }
+
+  function readElementGeom(id: string): GeomStreamEntry[] & { viewIndependent: boolean } {
```

Comment:
why not wrap the array in an object: `{entries: GeomStreamEntry[]; viewIndependent: boolean}` instead of patch it with a property? Seems like this will lead to some awkward `as any` casts down the line

Reply:
More annoying to iterate. Only care about the flag in one test.